src/examples/menu/Button.tsx

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { useState } from 'react';
+import { Col, Row } from '@zendeskgarden/react-grid';
+import { Button } from '@zendeskgarden/react-buttons';
+import { Menu, Item } from '@zendeskgarden/react-dropdowns.next';
+import { ReactComponent as ChevronIcon } from '@zendeskgarden/svg-icons/src/16/chevron-down-stroke.svg';
+
+const Example = () => {
+  const [rotated, setRotated] = useState<boolean | undefined>();
+
+  const handleChange = ({ isExpanded }: { isExpanded?: boolean }) => {
+    if (isExpanded !== undefined) {
+      setRotated(isExpanded);
+    }
+  };
+
+  return (
+    <Row justifyContent="center">
+      <Col textAlign="center">
+        <Menu
+          button={props => (
+            <Button {...props}>
+              Menu
+              <Button.EndIcon isRotated={rotated}>
+                <ChevronIcon />
+              </Button.EndIcon>
```

Comment:
Dang. It just now occurs to me that our `Menu` component should be doing this out of the box. We might need to deliver an enhancement.

Reply:
Hmmm, the legacy menu wasn't doing this so I presumed we would be the same here. But it's not difficult to implement, so definitely fair game to add in.

Reply:
Will uptake new Menu version when released this week - I'll modify this example to demonstrate custom `button` another way so there's no conflict.